File: mobile/lib/api/models/representations/queueable_music.dart
import "package:wavelength/api/models/embedded.dart";
import "package:wavelength/api/models/enums/video_type.dart";

class QueueableMusic {
  final String videoId;
  final String title;
  final String thumbnail;
  final List<EmbeddedArtist> artists;
  final EmbeddedAlbum? album;
  final VideoType videoType;

  const QueueableMusic({
    required this.videoId,
    required this.title,
    required this.thumbnail,
    required this.artists,
    required this.album,
    required this.videoType,
  });
}

File: server/services/player/models/queueable_music.go
package models

type QueueableMusic struct {
	VideoID   string `json:"videoId"`
	Title     string `json:"title"`
	Thumbnail string `json:"thumbnail"`
	Duration  any    `json:"duration"`
	VideoType string `json:"videoType"`
	Artists   []struct {
		Title    string `json:"title"`
		BrowseID string `json:"browseId"`
	} `json:"artists"`
	Album *struct {
		Title    string `json:"title"`
		BrowseID string `json:"browseId"`
	} `json:"album,omitempty"`
}
